aws/resource_aws_vpn_gateway.go

```diff
@@ -153,29 +153,38 @@ func resourceAwsVpnGatewayDelete(d *schema.ResourceData, meta interface{}) error
 	}
 
 	log.Printf("[INFO] Deleting VPN gateway: %s", d.Id())
-
-	return resource.Retry(5*time.Minute, func() *resource.RetryError {
-		_, err := conn.DeleteVpnGateway(&ec2.DeleteVpnGatewayInput{
-			VpnGatewayId: aws.String(d.Id()),
-		})
+	input := &ec2.DeleteVpnGatewayInput{
+		VpnGatewayId: aws.String(d.Id()),
+	}
+	err := resource.Retry(5*time.Minute, func() *resource.RetryError {
+		_, err := conn.DeleteVpnGateway(input)
 		if err == nil {
 			return nil
 		}
 
-		ec2err, ok := err.(awserr.Error)
-		if !ok {
+		if isAWSErr(err, "InvalidVpnGatewayID.NotFound", "") {
+			return nil
+		}
+		if isAWSErr(err, "IncorrectState", "") {
 			return resource.RetryableError(err)
 		}
-
-		switch ec2err.Code() {
-		case "InvalidVpnGatewayID.NotFound":
-			return nil
-		case "IncorrectState":
+		if !isAWSErr(err, "", "") {
```

Comment:
This is not a valid error condition to test 😉 All AWS Go SDK errors should have an error code. We can just remove this `if` condition here.